### PR TITLE
ACR 8 cards

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/alexios_deimos_of_kosmos.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Name:Alexios, Deimos of Kosmos
+ManaCost:3 R
+Types:Legendary Creature Human Berserker
+PT:4/4
+K:Trample
+S:Mode$ CantAttack | ValidCard$ Card.Self | Target$ Player.CardOwner | Description$ CARDNAME can't attack its owner.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/alexios_deimos_of_kosmos.txt:6]: descriptions not matching

File: forge-gui/res/cardsfolder/upcoming/jacob_frye.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Name:Jacob Frye
+ManaCost:2 B
+Types:Legendary Creature Human Assassin
+PT:3/2
+K:Partner:Evie Frye
+T:Mode$ DamageDoneOnce | CombatDamage$ True | ValidSource$ Assassin.YouCtrl | ValidTarget$ Player | TriggerZones$ Battlefield | Execute$ TrigExile | TriggerDescription$ Whenever one or more Assassins you control deal combat damage to a player, exile up to one target Assassin card or card with freerunning from your graveyard. If you do, copy it. You may cast the copy.
+SVar:TrigExile:DB$ ChangeZone | ValidTgts$ Card.Assassin+YouCtrl,Card.withFreerunning+YouCtrl | Origin$ Graveyard | Destination$ Exile | TgtPrompt$ Select target creature card with lesser power in your graveyard | RememberChanged$ True | Optional$ True | SubAbility$ DBCast
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/jacob_frye.txt:7]: wrong targeting

File: forge-gui/res/cardsfolder/upcoming/kassandra_eagle_bearer.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Name:Kassandra, Eagle Bearer
+ManaCost:1 R W
+Types:Legendary Creature Human Assassin Warrior
+PT:2/2
+K:Haste
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigSearch | OptionalDecider$ You | TriggerDescription$ When NICKNAME enters the battlefield, search your graveyard, hand, and library for a card named The Spear of Leonidas, put it onto the battlefield, then shuffle.
+SVar:TrigSearch:DB$ ChangeZone | OriginAlternative$ Graveyard,Hand | Hidden$ True | Origin$ Library | Destination$ Battlefield | ChangeType$ Card.namedThe Spear of Leonidas  | ChangeNum$ 1 | ShuffleNonMandatory$ True
+T:Mode$ DamageDone | ValidSource$ Creature.YouCtrl+EquippedBy Equipment.YouCtrl+Legendary | ValidTarget$ Player | CombatDamage$ True | TriggerZones$ Battlefield | Execute$ TrigDraw | TriggerDescription$ Whenever a creature you control with a legendary Equipment attached to it deals combat damage to a player, draw a card.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/kassandra_eagle_bearer.txt:8]: wrong valid

File: forge-gui/res/cardsfolder/upcoming/loyal_inventor.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+Name:Loyal Inventor
+ManaCost:2 U
+Types:Creature Human Artificer
+PT:2/3
+K:Vigilance
+T:Mode$ ChangesZone | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigTutor | TriggerDescription$ When CARDNAME enters the battlefield, you may search your library for an artifact card, reveal it, then shuffle. Put that card into your hand if you control an Assassin. Otherwise, put that card on top of your library.
+SVar:TrigTutor:DB$ ChangeZone | Origin$ Library | Destination$ Library | ChangeType$ Artifact | ChangeNum$ 1 | Reveal$ True | RememberChanged$ True | SubAbility$ DBChangeZone
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/loyal_inventor.txt:7]: "may"

File: forge-gui/res/cardsfolder/upcoming/loyal_inventor.txt [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+SVar:DBChangeZone:DB$ ChangeZone | Defined$ Remembered | Origin$ Library | Destination$ Hand | ConditionCheckSVar$ X | SubAbility$ DBChangeZone2
+SVar:DBChangeZone2:DB$ ChangeZone | Defined$ Remembered | Origin$ Library | ConditionCheckSVar$ X |  ConditionSVarCompare$ LE0 | Destination$ Library | LibraryPosition$ 0 | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+SVar:X:Count$Valid Card.Assassin
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/loyal_inventor.txt:11]: wrong valid

File: forge-gui/res/cardsfolder/upcoming/shaun_and_rebecca_agents.txt [hunk 6]
```diff
@@ -0,0 +1,13 @@
+Name:Shaun & Rebecca, Agents
+ManaCost:1 G W U
+Types:Legendary Creature Human Assassin Scientist
+PT:4/4
+K:Vigilance
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigSearch | OptionalDecider$ You | TriggerDescription$ When CARDNAME enters the battlefield, search your graveyard, hand, and library for a card named The Animus and put it onto the battlefield, then shuffle.
+SVar:TrigSearch:DB$ ChangeZone | OriginAlternative$ Graveyard,Hand | Hidden$ True | Origin$ Library | Destination$ Battlefield | ChangeType$ Card.namedThe Animus  | ChangeNum$ 1 | ShuffleNonMandatory$ True
+A:AB$ Mana | Cost$ T |  Produced$ C | SubAbility$ DBImmediateTrig | SpellDescription$ Add {C}. When you do, mill two cards.
+SVar:DBImmediateTrig:DB$ ImmediateTrigger | Execute$ TrigMill | SubAbility$ DBCleanup | TriggerDescription$ When you do, mill two cards.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/shaun_and_rebecca_agents.txt:9]: bad sub

File: forge-gui/res/cardsfolder/upcoming/sokrates_athenian_teacher.txt [hunk 7]
```diff
@@ -0,0 +1,12 @@
+Name:Sokrates, Athenian Teacher
+ManaCost:1 W U
+Types:Legendary Creature Human Advisor
+PT:0/4
+K:Defender
+S:Mode$ Continuous | Affected$ Card.Self+untapped | AddKeyword$ Hexproof | Description$ CARDNAME has hexproof as long as it's untapped.
+A:AB$ Effect  | Cost$ T | ValidTgts$ Creature | PrecostDesc$ Sokratic Dialogue | ReplacementEffects$ RPrevent | RememberObjects$ Targeted | SpellDescription$ Until end of turn, target creature gains "If this creature would deal combat damage to a player, prevent that damage. This creature's controller and that player each draw half that many cards, rounded down."
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/sokrates_athenian_teacher.txt:7]: why would "gains" be Effect?

File: forge-gui/res/cardsfolder/upcoming/jacob_frye.txt (hunk 2 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/jacob_frye.txt]: still not optional

File: forge-gui/res/cardsfolder/upcoming/alexios_deimos_of_kosmos.txt (hunk 1 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/alexios_deimos_of_kosmos.txt]: why empty lines now?

File: forge-gui/res/cardsfolder/upcoming/sokrates_athenian_teacher.txt (hunk 7 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/sokrates_athenian_teacher.txt]: the Draw line needs changing too

File: forge-gui/res/cardsfolder/upcoming/sokrates_athenian_teacher.txt [hunk 8]
```diff
@@ -0,0 +1,12 @@
+Name:Sokrates, Athenian Teacher
+ManaCost:1 W U
+Types:Legendary Creature Human Advisor
+PT:0/4
+K:Defender
+S:Mode$ Continuous | Affected$ Card.Self+untapped | AddKeyword$ Hexproof | Description$ CARDNAME has hexproof as long as it's untapped.
+A:AB$ Animate | Cost$ T | ValidTgts$ Creature | PrecostDesc$ Sokratic Dialogue | Replacements$ RPrevent | SpellDescription$ Until end of turn, target creature gains "If this creature would deal combat damage to a player, prevent that damage. This creature's controller and that player each draw half that many cards, rounded down."
+SVar:RPrevent:Event$ DamageDone | ValidTarget$ Player | Prevent$ True | IsCombat$ True | ValidSource$ Card.Self | ReplaceWith$ Draw | Description$ If this creature would deal combat damage to a player, prevent that damage. This creature's controller and that player each draw half that many cards, rounded down.
+SVar:Draw:DB$ Draw | Defined$ ReplacedSourceController & EffectSourceController | NumCards$ X 
+SVar:X:ReplaceCount$DamageAmount
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/sokrates_athenian_teacher.txt:10]: wrong amount

File: forge-gui/res/cardsfolder/upcoming/shaun_and_rebecca_agents.txt [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+ManaCost:1 G W U
+Types:Legendary Creature Human Assassin Scientist
+PT:4/4
+K:Vigilance
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigSearch | OptionalDecider$ You | TriggerDescription$ When CARDNAME enters the battlefield, search your graveyard, hand, and library for a card named The Animus and put it onto the battlefield, then shuffle.
+SVar:TrigSearch:DB$ ChangeZone | OriginAlternative$ Graveyard,Hand | Hidden$ True | Origin$ Library | Destination$ Battlefield | ChangeType$ Card.namedThe Animus  | ChangeNum$ 1 | ShuffleNonMandatory$ True
+A:AB$ Mana | Produced$ C | SubAbility$ DBImmediateTrig | SpellDescription$ Add {C}. When you do, mill two cards.
+SVar:DBImmediateTrig:DB$ ImmediateTrigger | Execute$ TrigMill | SubAbility$ DBCleanup | TriggerDescription$ When you do, mill two cards.
+SVar:TrigMill:DB$ Mill | NumCards$ 2 | ValidTgts$ You
+DeckHints:Name$he Animus
```
Comment (squee1968) [inline forge-gui/res/cardsfolder/upcoming/shaun_and_rebecca_agents.txt:11]: Isn't this supposed to be "The Animus"?

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/shaun_and_rebecca_agents.txt:11]: Thanks fixed

File: forge-gui/res/cardsfolder/upcoming/alexios_deimos_of_kosmos.txt [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+ManaCost:3 R
+Types:Legendary Creature Human Berserker
+PT:4/4
+K:Trample
+S:Mode$ CantAttack | ValidCard$ Card.Self | Target$ Player.CardOwner | Description$ CARDNAME attacks each combat if able, can't be sacrificed, and can't attack its owner.
+S:Mode$ MustAttack | ValidCreature$ Card.Self | Secondary$ True | Description$ CARDNAME attacks each combat if able, can't be sacrificed, and can't attack its owner.
+S:Mode$ CantSacrifice | ValidCard$ Card.Self | Secondary$ True | Description$ CARDNAME attacks each combat if able, can't be sacrificed, and can't attack its owner.
+T:Mode$ Phase | Phase$ Upkeep | ValidPlayer$ Player | TriggerZones$ Battlefield | Execute$ TrigControl | TriggerDescription$ At the beginning of each player's upkeep, that player gains control of NICKNAME, untaps it, and puts a +1/+1 counter on it. It gains haste until end of turn.
+SVar:TrigControl:DB$ GainControl | Defined$ Self | NewController$ TriggeredPlayer | Untap$ True | SubAbility$ DBPutCounter
+SVar:DBPutCounter:DB$ PutCounter | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1 | SubAbility$ DBPump
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/alexios_deimos_of_kosmos.txt:11]: needs Placer$

File: forge-gui/res/cardsfolder/upcoming/sokrates_athenian_teacher.txt [hunk 11]
```diff
@@ -0,0 +1,12 @@
+Name:Sokrates, Athenian Teacher
+ManaCost:1 W U
+Types:Legendary Creature Human Advisor
+PT:0/4
+K:Defender
+S:Mode$ Continuous | Affected$ Card.Self+untapped | AddKeyword$ Hexproof | Description$ CARDNAME has hexproof as long as it's untapped.
+A:AB$ Animate | Cost$ T | ValidTgts$ Creature | PrecostDesc$ Sokratic Dialogue | Replacements$ RPrevent | SpellDescription$ Until end of turn, target creature gains "If this creature would deal combat damage to a player, prevent that damage. This creature's controller and that player each draw half that many cards, rounded down."
+SVar:RPrevent:Event$ DamageDone | ValidTarget$ Player | Prevent$ True | IsCombat$ True | ValidSource$ Card.Self | ReplaceWith$ Draw | Description$ If this creature would deal combat damage to a player, prevent that damage. This creature's controller and that player each draw half that many cards, rounded down.
+SVar:Draw:DB$ Draw | Defined$ ReplacedSourceController & CardController | NumCards$ X
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/sokrates_athenian_teacher.txt:9]: isn't that the same player twice?

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/sokrates_athenian_teacher.txt:9]: Yeah, any ideas about the correct defined for "that player" ?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/sokrates_athenian_teacher.txt:9]: Why not just "ReplacedTarget"?

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/sokrates_athenian_teacher.txt:9]: Thanks working now ! 
image

**Pull request merged**